### ISSUE TITLE
[check-update] get changed dataId error, code: 403

### ISSUE THREAD

**Issue by gongmb** [alibaba/nacos#3502]
当修改nacos.core.auth.enabled=false时 可以注册也可以获取到配置管理的配置
当修改nacos.core.auth.enabled=true时 ,服务列表可以注册进来,但是配置管理面的配置获取不到
一直报[check-update] get changed dataId error, code: 403,项目中加了一下配置还是不行,有没有大佬遇到过类似的问题
spring.cloud.nacos.config.username=nacos
spring.cloud.nacos.config.password=nacos

版本
  <groupId>com.alibaba.cloud</groupId>
  <artifactId>spring-cloud-starter-alibaba-nacos-config</artifactId>
  <version>2.2.1.RELEASE</version>

**Comment by KomachiSion** [comment]
@gongmb What's your nacos server version

**Comment by gongmb** [comment]
> @gongmb您的nacos服务器版本是多少

nacos 版本是1.2.1,而且发现nacos管理平台使用只读账号登录也可以进行配置的修改

**Comment by gongmb** [comment]
Nacos 开启认证后,sentinel-datasource无法访问nacos,所以一直报错

**Comment by sujianyu** [comment]
> Nacos 开启认证后,sentinel-datasource无法访问nacos,所以一直报错

然后怎么解决啊？

**Comment by hegphegp** [comment]
* 现在最新版本是1.4.0，开启权限认证后，一直抛这个错，控制台一直在抛错，怎么解决？把权限认证一关，就不抛错了。这个算是生产线上的严重事故吗？
```
nacos-server:1.4.0
com.alibaba.cloud:spring-cloud-starter-alibaba-nacos-config:2.2.3.RELEASE
com.alibaba.cloud:spring-cloud-starter-alibaba-nacos-discovery:2.2.3.RELEASE
```
* 配置信息
```
spring:
  application:
    name: test
  cloud:
    nacos:
      username: nacos
      password: nacos
      discovery:
        username: nacos
        password: nacos
        server-addr: localhost:8848
        service: ${spring.application.name}
        group: DEFAULT_GROUP
        namespace: dev1
        enabled: true
      config:
        username: nacos
        password: nacos
        enabled: true
        server-addr: ${spring.cloud.nacos.discovery.server-addr}
        prefix: ${spring.application.name}
        file-extension: yaml
        refresh-enabled: true
        group: ${spring.cloud.nacos.discovery.group}
        namespace: ${spring.cloud.nacos.discovery.namespace}
```

```
2020-12-22 17:02:15.612 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:15.810 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:15.811 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:15.811 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:16.008 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:16.008 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:16.008 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:16.210 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:16.211 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:16.211 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:16.410 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:16.411 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:16.412 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:16.610 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:16.610 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:16.610 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:16.803 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:16.803 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:16.803 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:17.013 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:17.014 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:17.014 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
2020-12-22 17:02:17.213 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
2020-12-22 17:02:17.214 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=testDEFAULT_GROUPtest.propertiesDEFAULT_GROUP}
2020-12-22 17:02:17.214 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
```

**Comment by hzauitdog** [comment]
> @gongmb What's your nacos server version



> * 现在最新版本是1.4.0，开启权限认证后，一直抛这个错，控制台一直在抛错，怎么解决？把权限认证一关，就不抛错了。这个算是生产线上的严重事故吗？
> 
> ```
> nacos-server:1.4.0
> com.alibaba.cloud:spring-cloud-starter-alibaba-nacos-config:2.2.3.RELEASE
> com.alibaba.cloud:spring-cloud-starter-alibaba-nacos-discovery:2.2.3.RELEASE
> ```
> 
> * 配置信息
> 
> ```
> spring:
>   application:
>     name: test
>   cloud:
>     nacos:
>       username: nacos
>       password: nacos
>       discovery:
>         username: nacos
>         password: nacos
>         server-addr: localhost:8848
>         service: ${spring.application.name}
>         group: DEFAULT_GROUP
>         namespace: dev1
>         enabled: true
>       config:
>         username: nacos
>         password: nacos
>         enabled: true
>         server-addr: ${spring.cloud.nacos.discovery.server-addr}
>         prefix: ${spring.application.name}
>         file-extension: yaml
>         refresh-enabled: true
>         group: ${spring.cloud.nacos.discovery.group}
>         namespace: ${spring.cloud.nacos.discovery.namespace}
> ```
> 
> ```
> 2020-12-22 17:02:15.612 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:15.810 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:15.811 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:15.811 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:16.008 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:16.008 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:16.008 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:16.210 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:16.211 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:16.211 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:16.410 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:16.411 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:16.412 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:16.610 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:16.610 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:16.610 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:16.803 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:16.803 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:16.803 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:17.013 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:17.014 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:17.014 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> 2020-12-22 17:02:17.213 ERROR 22215 --- [-localhost_8848] c.a.n.client.config.impl.ClientWorker    : [fixed-localhost_8848] [check-update] get changed dataId error, code: 403
> 2020-12-22 17:02:17.214 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : HTTP method: POST, url: http://localhost:8848/nacos/v1/cs/configs/listener, body: {Listening-Configs=test�DEFAULT_GROUP��test.properties�DEFAULT_GROUP��}
> 2020-12-22 17:02:17.214 DEBUG 22215 --- [-localhost_8848] c.a.n.c.c.impl.ConfigHttpClientManager   : Execute via interceptors :[com.alibaba.nacos.client.config.impl.ConfigHttpClientManager$LimiterHttpClientRequestInterceptor@25d3cfc8]
> ```

你们怎么解决这个问题的

**Comment by hzauitdog** [comment]
我发现自己集成了seata，由于seata客户端没配密码所以报错

seata:
  registry:
    nacos:
      group: DEFAULT_GROUP
      server-addr: ${spring.cloud.nacos.discovery.server-addr}
    type: nacos
  config:
    type: nacos
    nacos:
      server-addr: ${spring.cloud.nacos.config.server-addr}
      group: SEATA_GROUP
  tx-service-group: abc-service-group
  enable-auto-data-source-proxy: true

**Comment by hzauitdog** [comment]
``seata:
  registry:
    nacos:
      username: nacos
      password: nacos
      group: DEFAULT_GROUP
      server-addr: ${spring.cloud.nacos.discovery.server-addr}
    type: nacos
  config:
    type: nacos
    nacos:
      server-addr: ${spring.cloud.nacos.config.server-addr}
      group: SEATA_GROUP
  tx-service-group: abc-service-group
  enable-auto-data-source-proxy: true

**Comment by Dayan0000** [comment]
cloud下除了nacos 还有什么组件？
比如sentinel enabled: false或者依赖注释掉都能解决

**Comment by hbcolorful** [comment]
> * 现在最新版本是1.4.0，开启权限认证后，一直抛这个错，控制台一直在抛错，怎么解决？把权限认证一关，就不抛错了。这个算是生产线上的严重事故吗？

请问这问题有解决办法吗？

**Comment by hbcolorful** [comment]
> cloud下除了nacos 还有什么组件？ 比如sentinel enabled: false或者依赖注释掉都能解决
请问下还有其他的原因吗?我这只引入了服务注册、发现的依赖，同样存在403的问题。

**错误日志信息：**

```
2023-11-23 11:27:24.363 ERROR [dv-file-center,,,] 15968 --- [           main] c.a.n.client.config.impl.ClientWorker    : [fixed-127.0.0.1_8849] [sub-server-error] no right, dataId=dv-file-center, group=discovery-guide-group, tenant=
2023-11-23 11:27:24.364 ERROR [dv-file-center,,,] 15968 --- [.127.0.0.1_8849] c.a.n.client.config.impl.ClientWorker    : [fixed-127.0.0.1_8849] [check-update] get changed dataId error, code: 403
2023-11-23 11:27:24.366  WARN [dv-file-center,,,] 15968 --- [           main] c.n.d.p.c.initializer.ConfigInitializer  : Get remote config list failed
2023-11-23 11:27:24.473 ERROR [dv-file-center,,,] 15968 --- [.127.0.0.1_8849] c.a.n.client.config.impl.ClientWorker    : [fixed-127.0.0.1_8849] [check-update] get changed dataId error, code: 403
```

**依赖版本**

```
		<dependency>
            <groupId>com.alibaba.cloud</groupId>
            <artifactId>spring-cloud-starter-alibaba-nacos-discovery</artifactId>
            <version>2.2.6.RELEASE</version>
            <exclusions>
                <exclusion>
                    <groupId>com.alibaba.nacos</groupId>
                    <artifactId>nacos-client</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.alibaba.cloud</groupId>
            <artifactId>spring-cloud-starter-alibaba-nacos-config</artifactId>
            <version>2.2.6.RELEASE</version>
            <exclusions>
                <exclusion>
                    <groupId>com.alibaba.nacos</groupId>
                    <artifactId>nacos-client</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.alibaba.nacos</groupId>
            <artifactId>nacos-client</artifactId>
            <version>1.4.6</version>
        </dependency>
```

**Comment by jiafaxin** [comment]
[check-update] get changed dataId error, code: 504 这是什么情况，我客户端显示已经注册到nacos
